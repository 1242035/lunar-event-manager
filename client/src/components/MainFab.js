import React from 'react';

/**
 * This is the bigger fab that expands/collapses the smaller fabs.
 * @param{Object} props - passed properties.
 * @return{React.Component}
 */
const MainFab = (props) => {
  // Hide fab if user not logged in.
  const hide = localStorage.getItem('user.token') && localStorage.getItem('user.token') !== 'undefined' && localStorage.getItem('user.token') !== '' ? '' : 'hide';

  return (
    <div id="main-fab" className={`io-fab ${hide}`} onClick={props.toggleFabGroup}><i className="fa fa-plus" /></div>
  );
};

export default MainFab;
