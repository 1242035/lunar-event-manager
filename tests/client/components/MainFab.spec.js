import React from 'react';
import { shallow } from 'enzyme';
import MainFab from '../../../client/src/components/MainFab';

describe('MainFab', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<MainFab />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<MainFab />);
    expect(wrapper.length).toEqual(1);
  });
});
