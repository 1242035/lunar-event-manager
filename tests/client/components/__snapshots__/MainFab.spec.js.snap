// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFab Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainFab />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="fa fa-plus"
      />,
      "className": "io-fab hide",
      "id": "main-fab",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fa fa-plus",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="fa fa-plus"
        />,
        "className": "io-fab hide",
        "id": "main-fab",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-plus",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
